src/nvim/api/tabpage.c

```diff
@@ -122,6 +122,46 @@ Window nvim_tabpage_get_win(Tabpage tabpage, Error *err)
   abort();
 }
 
+/// Sets the current window in a tabpage
+///
+/// @param tabpage  Tabpage handle, or 0 for current tabpage
+/// @param win Window handle
+/// @param[out] err Error details, if any
+/// @return Window handle
+void nvim_tabpage_set_win(Tabpage tabpage, Window win, Error *err)
+  FUNC_API_SINCE(12)
+{
+  tabpage_T *tab = find_tab_by_handle(tabpage, err);
+  if (!tab) {
+    return;
+  }
+
+  win_T *wp = find_window_by_handle(win, err);
+  if (!wp) {
+    return;
+  }
+
+  tabpage_T *win_tp = win_find_tabpage(wp);
+
+  if (win_tp == NULL || win_tp->handle != tab->handle) {
```

Comment:
Do I need this null check? It looks like `get_win` assumes that there will always be a tabpage for a window.